File: VICSApp/CoreSystem/DHCD/FormUI/Meeting/Candidates_ins_update.vb
Public Class Candidates_ins_update
    Dim controlcode As String
    Dim updateelectioncode As Integer
    Dim updatecandidatecode As Integer
    Public Sub New(ByVal controlcode As String, ByVal electioncode As Integer, ByVal candidatecode As Integer)
        Me.controlcode = controlcode
        Me.updateelectioncode = electioncode
        Me.updatecandidatecode = candidatecode
        ' This call is required by the Windows Form Designer.
        InitializeComponent()
        ' Add any initialization after the InitializeComponent() call.

    End Sub
    Private Sub Candidates_ins_update_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        MaskedTextBox1.Text = Mainform.workingmeeting
        If controlcode = "Update" Then
            NumericUpDown1.Value = Me.updateelectioncode
            NumericUpDown1.ReadOnly = True
            NumericUpDown2.Value = Me.updatecandidatecode
            NumericUpDown2.ReadOnly = True
            Button1.Text = "Cập nhật"
            Me.Text = "Cập nhật thông tin ứng viên"
            Dim dt As New DataTable
            Try
                dt = Mainform.BenlyDal.Candidates_getlist(Mainform.workingmeeting, updateelectioncode, updatecandidatecode)
            Catch ex As Exception
                MsgBox("lỗi" + ex.Message)
                Exit Sub
            End Try
            MaskedTextBox3.Text = dt.Rows(0).Item("Electionname")
            MaskedTextBox2.Text = dt.Rows(0).Item("Candidatename")
            MaskedTextBox4.Text = dt.Rows(0).Item("Candidateaddress")
        End If
    End Sub

    Private Sub NumericUpDown1_ValueChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles NumericUpDown1.ValueChanged
        Dim dt As New DataTable
        Try
            dt = Mainform.BenlyDal.Election_getlist(Mainform.workingmeeting, NumericUpDown1.Value)
        Catch ex As Exception
            MsgBox("lỗi" + ex.Message)
            Exit Sub
        End Try
        If dt.Rows.Count = 1 Then
            MaskedTextBox3.Text = dt.Rows(0).Item("electionname")
        Else
            MaskedTextBox3.Text = ""
        End If
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        If controlcode = "Add" Then
            Try
                Mainform.BenlyDal.Candidate_insert(Mainform.workingmeeting, NumericUpDown1.Value, NumericUpDown2.Value, MaskedTextBox2.Text, MaskedTextBox4.Text)
                Me.Close()
            Catch ex As Exception
                MsgBox("Lỗi : " + ex.Message)
            End Try
        ElseIf controlcode = "Update" Then
            Try
                Mainform.BenlyDal.Candidate_update(Mainform.workingmeeting, NumericUpDown1.Value, NumericUpDown2.Value, MaskedTextBox2.Text, MaskedTextBox4.Text)
                Me.Close()
            Catch ex As Exception
                MsgBox("Lỗi : " + ex.Message)
            End Try

        End If
    End Sub

    Private Sub Button3_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button3.Click
        Me.Close()
    End Sub
End Class

File: VICSApp/CoreSystem/DHCD/FormUI/Meeting/Candidates_ins_update.cs
using System;
using System.Data;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace pmDHCD
{
    public partial class Candidates_ins_update
    {
        private string controlcode;
        private int updateelectioncode;
        private int updatecandidatecode;

        public Candidates_ins_update(string controlcode, int electioncode, int candidatecode)
        {
            this.controlcode = controlcode;
            updateelectioncode = electioncode;
            updatecandidatecode = candidatecode;
            // This call is required by the Windows Form Designer.
            InitializeComponent();
            _NumericUpDown1.Name = "NumericUpDown1";
            _Button3.Name = "Button3";
            _Button1.Name = "Button1";
            // Add any initialization after the InitializeComponent() call.

        }

        private void Candidates_ins_update_Load(object sender, EventArgs e)
        {
            MaskedTextBox1.Text = My.MyProject.Forms.Mainform.workingmeeting;
            if (controlcode == "Update")
            {
                NumericUpDown1.Value = updateelectioncode;
                NumericUpDown1.ReadOnly = true;
                NumericUpDown2.Value = updatecandidatecode;
                NumericUpDown2.ReadOnly = true;
                Button1.Text = "Cập nhật";
                Text = "Cập nhật thông tin ứng viên";
                var dt = new DataTable();
                try
                {
                    dt = My.MyProject.Forms.Mainform.BenlyDal.Candidates_getlist(My.MyProject.Forms.Mainform.workingmeeting, updateelectioncode, updatecandidatecode);
                }
                catch (Exception ex)
                {
                    Interaction.MsgBox("lỗi" + ex.Message);
                    return;
                }

                MaskedTextBox3.Text = Conversions.ToString(dt.Rows[0]["Electionname"]);
                MaskedTextBox2.Text = Conversions.ToString(dt.Rows[0]["Candidatename"]);
                MaskedTextBox4.Text = Conversions.ToString(dt.Rows[0]["Candidateaddress"]);
            }
        }

        private void NumericUpDown1_ValueChanged(object sender, EventArgs e)
        {
            var dt = new DataTable();
            try
            {
                dt = My.MyProject.Forms.Mainform.BenlyDal.Election_getlist(My.MyProject.Forms.Mainform.workingmeeting, NumericUpDown1.Value);
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("lỗi" + ex.Message);
                return;
            }

            if (dt.Rows.Count == 1)
            {
                MaskedTextBox3.Text = Conversions.ToString(dt.Rows[0]["electionname"]);
            }
            else
            {
                MaskedTextBox3.Text = "";
            }
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            if (controlcode == "Add")
            {
                try
                {
                    My.MyProject.Forms.Mainform.BenlyDal.Candidate_insert(My.MyProject.Forms.Mainform.workingmeeting, NumericUpDown1.Value, NumericUpDown2.Value, MaskedTextBox2.Text, MaskedTextBox4.Text);
                    Close();
                }
                catch (Exception ex)
                {
                    Interaction.MsgBox("Lỗi : " + ex.Message);
                }
            }
            else if (controlcode == "Update")
            {
                try
                {
                    My.MyProject.Forms.Mainform.BenlyDal.Candidate_update(My.MyProject.Forms.Mainform.workingmeeting, NumericUpDown1.Value, NumericUpDown2.Value, MaskedTextBox2.Text, MaskedTextBox4.Text);
                    Close();
                }
                catch (Exception ex)
                {
                    Interaction.MsgBox("Lỗi : " + ex.Message);
                }
            }
        }

        private void Button3_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}